dts/xtensa/nxp/nxp_imx8m.dtsi

```diff
@@ -33,6 +33,15 @@
 	};
 
 	soc {
+		interrupt-parent = <&irqsteer>;
+
+		irqsteer: interrupt-controller@ffffffff {
+			compatible = "nxp,irqsteer-intc";
+			interrupt-controller;
+			reg=<0xffffffff 0xffffffff>;
+			#interrupt-cells = <2>;
+		};
```

Comment:
I'm a bit lost. Is this a working sample without an interrupt controller?

Reply:
Yes, DSP has a direct interrupt line from MU.
You can check reference manual [here](https://www.nxp.com/products/processors-and-microcontrollers/arm-processors/i-mx-applications-processors/i-mx-8-applications-processors/i-mx-8m-plus-arm-cortex-a53-machine-learning-vision-multimedia-and-industrial-iot:IMX8MPLUS?cid=ad_pro10955341_tac1219403_aac&gclid=CjwKCAjwtuOlBhBREiwA7agf1jtzY5Dk5rD4iEIg0MKO9teAQ1pPg4qYIFvNdk5tKStWNFI_f9PlWBoC2wwQAvD_BwE) - section 7.1.4 Audio DSP Interrupts (page 1003)

Reply:
I got 404 at that link.

Reply:
Sorry, don't know what happend.
Here's a snippet:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/74421316/9032a13d-46e6-4dea-87fe-4e9efe807052)

Reply:
Sorry, lost again, I'm not familiar with this architecture. There is no interrupt vector table for the DSP? How the interrupts are dispatched to the correct ISR on this platform?

Reply:
Interrupt line from MU is 7 (from the above table). This is selected here: https://github.com/zephyrproject-rtos/zephyr/pull/60455/files#diff-a35135d3554f4ea1b952169f2deb76ead643a1a72b7342cf7aa4cc9b9019d747R75

> There is no interrupt vector table for the DSP?

This is generated on build:
```
/* AUTO-GENERATED by gen_isr_tables.py, do not edit! */

#include <zephyr/toolchain.h>
#include <zephyr/linker/sections.h>
#include <zephyr/sw_isr_table.h>
#include <zephyr/arch/cpu.h>

typedef void (* ISR)(const void *);
struct _isr_table_entry __sw_isr_table _sw_isr_table[32] = {
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)0x92404bd0},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x92408124, (ISR)0x924049a0},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
	{(const void *)0x0, (ISR)((uintptr_t)&z_irq_spurious)},
};
```
> How the interrupts are dispatched to the correct ISR on this platform?

See: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/xtensa/nxp_adsp/imx8m/include/_soc_inthandlers.h

MU, timer0 and timer1 are Level1 interrupts: https://docs.zephyrproject.org/latest/kernel/services/interrupts.html#multi-level-interrupt-handling

Reply:
I was thinking something like:
```
	irqsteer: interrupt-controller@0 {
			compatible = "zephyr,dummy-intc";
			interrupt-controller;
			#interrupt-cells = <2>;
       };
```
with a `zephyr,dummy-intc.yaml`:
```
description: Dummy Interrupt Controller

compatible: "zephyr,intc"

include: [interrupt-controller.yaml, base.yaml]

properties:
  "#interrupt-cells":
    const: 2

interrupt-cells:
  - irq
  - priority
```
Not even the dummy driver is needed.